### PR TITLE
[ISSUE #3084] Reliance on default encoding[FileLoad]

### LINKED ISSUE #3084: [Enhancement] Reliance on default encoding[FileLoad]
### Search before asking

- [X] I had searched in the [issues](https://github.com/apache/eventmesh/issues?q=is%3Aissue) and found no similar issues.

### Enhancement Request

image
located at:
eventmesh-common/src/main/java/org/apache/eventmesh/common/config/FileLoad.java line 71,73
analysis and explanation:
1.Found a call to a method which will perform a byte to String (or String to byte) conversion, and will assume that the default platform encoding is suitable. This will cause the application behaviour to vary between platforms. Use an alternative API and specify a charset name or Charset object explicitly.
2. The method creates an IO stream object, does not assign it to any fields, pass it to other methods that might close it, or return it, and does not appear to close the stream on all paths out of the method.  This may result in a file descriptor leak.  It is generally a good idea to use a finally block to ensure that streams are closed.

### Describe the solution you'd like

1.Use try-with-resources to manage resources.
2. Use an alternative API and specify a charset name or Charset object explicitly.

### Are you willing to submit PR?

- [ ] Yes I am willing to submit a PR!

### PR DESCRIPTION
**Pull request by sanskar-thakur**
Fixes #3084 .

### Modifications

1. Specified charset with input-stream-reader API.
2. Used try-with-resources and handled io stream close.

### Documentation

- Does this pull request introduce a new feature? **(no)**
- If yes, how is the feature documented? **(not applicable)**
- If a feature is not applicable for documentation, explain why? **(bug fix)**
- If a feature is not documented yet in this PR, please create a followup issue for adding the documentation **(not applicable)**

### CODE REVIEW

**Comment by xwm1992** [comment]
image
@sanskar-thakur please fix the ci check error, take care of package import order.

**Review by mxsm — COMMENTED** [review]
@sanskar-thakur  When you're done you can check the format using the checkStyle plugin or running Gradle's checkstyleMain or checkstyleTest commands.

File: eventmesh-common/src/main/java/org/apache/eventmesh/common/config/FileLoad.java [hunk 1]
```diff
@@ -68,9 +72,28 @@ class PropertiesFileLoad implements FileLoad {
...
             if (StringUtils.isNotBlank(configInfo.getResourceUrl())) {
-                properties.load(new BufferedReader(new InputStreamReader(getClass().getResourceAsStream(configInfo.getResourceUrl()))));
+                try (InputStreamReader reader = new InputStreamReader(
+                    Objects.requireNonNull(getClass().getResourceAsStream(configInfo.getResourceUrl())), Constants.DEFAULT_CHARSET)) {
+                    properties.load(new BufferedReader(reader));
+                }
             } else {
-                properties.load(new BufferedReader(new FileReader(configInfo.getFilePath())));
+                FileReader reader = null;
+                try {
```
Comment (Alonexc) [inline eventmesh-common/src/main/java/org/apache/eventmesh/common/config/FileLoad.java:81]: Here you can also change to use try-with-resources, as well as specify the character set.
I have a suggestion, how about you look at:
image

Comment (sanskar-thakur) [reply eventmesh-common/src/main/java/org/apache/eventmesh/common/config/FileLoad.java]: Agreed. Updated this code block with try-with-resources.

**Review by Alonexc — APPROVED** [review]
LGTM

**Review by xwm1992 — APPROVED** [review]

**Pull request merged**